src/test/groovy/org/shipkit/internal/gradle/AutoDiscoverGradlePluginsPluginTest.groovy

```diff
@@ -0,0 +1,48 @@
+package org.shipkit.internal.gradle
+
+import com.gradle.publish.PluginBundleExtension
+import testutil.PluginSpecification
+
+class AutoDiscoverGradlePluginsPluginTest extends PluginSpecification {
+
+    public static final String META_INF_GRADLE_PLUGINS = 'src/main/resources/META-INF.gradle-plugins'
+
+    def "Apply"() {
+        expect:
+        project.plugins.apply(AutoDiscoverGradlePluginsPlugin)
+    }
+
+    def "DiscoverGradlePluginPropertyFiles"() {
+        given:
+        project.file(META_INF_GRADLE_PLUGINS).mkdirs()
+        project.file("$META_INF_GRADLE_PLUGINS/org.shipkit.test-plugin.properties") << "implementation-class=org.shipkit.TestPlugin"
+        when:
+        project.plugins.apply(AutoDiscoverGradlePluginsPlugin)
+        project.plugins.apply("com.gradle.plugin-publish")
+        then:
+        PluginBundleExtension pluginBundleExtension = project.getExtensions().findByType(PluginBundleExtension.class);
+        pluginBundleExtension.getPlugins()
+        pluginBundleExtension.getPlugins().size() == 1
+        pluginBundleExtension.plugins["testPlugin"]
+        pluginBundleExtension.plugins["testPlugin"].id == 'org.shipkit.test-plugin'
+    }
+
+    def "GeneratePluginName"() {
```

Comment:
This test covers the functionality of the previous test, too. I suggest to only keep this test. Less code to maintain!